src/shared/components/query/QueryStore.ts

```diff
@@ -728,47 +744,52 @@ export class QueryStore
-				genetic_profile_ids_PROFILE_PROTEIN_EXPRESSION: this.getSelectedProfileIdFromGeneticAlterationType("PROTEIN_LEVEL"),
-				Z_SCORE_THRESHOLD: this.zScoreThreshold,
-				RPPA_SCORE_THRESHOLD: this.rppaScoreThreshold,
-				data_priority: this.dataTypePriorityCode,
-				case_set_id: this.selectedSampleListId || '-1', // empty string won't work
-				case_ids: this.asyncCustomCaseSet.result.join('\r\n'),
-				gene_list: this.geneQuery || ' ', // empty string won't work
-				tab_index: this.forDownloadTab ? 'tab_download' : 'tab_visualize' as any,
-				transpose_matrix: this.transposeDataMatrix ? 'on' : undefined,
-				Action: 'Submit',
-			};
+	private getUrlParams(queryParams:CancerStudyQueryParams) {
+		let params: CancerStudyQueryUrlParams = {
+			cancer_study_id: (queryParams.selectedStudyIds.length === 1 ? queryParams.selectedStudyIds[0] : "all"),
+			genetic_profile_ids_PROFILE_MUTATION_EXTENDED: this.getSelectedProfileIdFromGeneticAlterationType("MUTATION_EXTENDED", queryParams.selectedProfileIds),
+			genetic_profile_ids_PROFILE_COPY_NUMBER_ALTERATION: this.getSelectedProfileIdFromGeneticAlterationType("COPY_NUMBER_ALTERATION", queryParams.selectedProfileIds),
+			genetic_profile_ids_PROFILE_MRNA_EXPRESSION: this.getSelectedProfileIdFromGeneticAlterationType("MRNA_EXPRESSION", queryParams.selectedProfileIds),
+			genetic_profile_ids_PROFILE_METHYLATION: this.getSelectedProfileIdFromGeneticAlterationType("METHYLATION", queryParams.selectedProfileIds) ||
+														this.getSelectedProfileIdFromGeneticAlterationType("METHYLATION_BINARY", queryParams.selectedProfileIds),
+			genetic_profile_ids_PROFILE_PROTEIN_EXPRESSION: this.getSelectedProfileIdFromGeneticAlterationType("PROTEIN_LEVEL", queryParams.selectedProfileIds),
+			Z_SCORE_THRESHOLD: queryParams.zScoreThreshold,
+			RPPA_SCORE_THRESHOLD: queryParams.rppaScoreThreshold,
+			data_priority: this.getDataTypePriorityCode(queryParams.dataTypePriority),
+			case_set_id: queryParams.selectedSampleListId || '-1', // empty string won't work
+			case_ids: queryParams.caseIds,
+			gene_list: queryParams.geneQuery || ' ', // empty string won't work
+			tab_index: this.forDownloadTab ? 'tab_download' : 'tab_visualize' as any,
+			transpose_matrix: this.transposeDataMatrix ? 'on' : undefined,
+			Action: 'Submit',
+		};
 
-			// Remove params with no value, because they may cause problems.
-			// For example, the server will always transpose if transpose_matrix is present, no matter the value.
-			for (let key in params)
-				if (!(params as any)[key])
-					delete (params as any)[key];
+		// Remove params with no value, because they may cause problems.
+		// For example, the server will always transpose if transpose_matrix is present, no matter the value.
+		for (let key in params)
+			if (!(params as any)[key])
+				delete (params as any)[key];

```

Comment:
Since we are already fixing the indentation here, can we also add the brackets for `for` and `if` blocks?